docs/tutorials/ad-hoc-simulation/planning-simulation.md

```diff
@@ -2,81 +2,110 @@
+Now you can start the ego vehicle driving by clicking the `Engage` button in `AutowareStatePanel`. Alteratively, you can manually engage the vehicle by running the following command:
 
-   ![restart-driving](images/planning/lane-following/restart-driving.png)
+```bash
+source ~/autoware/install/setup.bash
+ros2 topic pub /autoware/engage autoware_auto_vehicle_msgs/msg/Engage "engage: true" -1
+```
 
-   ![reach-goal](images/planning/lane-following/reach-goal.png)
+![start-driving](images/planning/lane-following/engage-and-start-planning.png)
 
-## How to simulate parking maneuvers
+### Parking maneuvers
 
 1. Set an initial pose and a goal pose, and engage the ego vehicle.
 
    ![after-set-goal-pose](images/planning/parking/after-set-goal-pose.png)
 
 2. When the vehicle approaches the goal, it will change to the parking mode.
+3. After that, the vehicle will start driving backwards and park at the destination parking spot.
 
-   ![change-to-parking-mode](images/planning/parking/change-to-parking-mode.png)
+   ![parking-maneuver](images/planning/parking/parking-maneuver.png)
 
-3. After that, the vehicle will start driving backwards and park at the destination parking spot.
+## Advanced Simulations
+
+### Dummy objects
+
+To place a dummy car/pedestrian, click the `2D Dummy Car` and/or `2D Dummy Pedestrian` button in the toolbar, and then set its pose by dragging on the map. You can set its velocity in `Tool Properties -> 2D Dummy Car/Pedestrian` panel. In the image the velocity is set to 0.
+
+![set-dummy-car](images/planning/lane-following/place-dummy-car.png)
+
+To delete any dummy objects placed in the view, click the `Delete All Objects` button in the toolbar.
+
+### Traffic light simulation
+
+#### Set traffic light
+
+To simulate traffic light recognition, go to `Panels -> Add new panel`, select `TrafficLightPublishPanel`, and then press `OK`. Then in the panel you need to set the `ID` and color of the traffic light.
+
+You can check the ID of each traffic light on Rviz. If the IDs do not pop up, select `Map/Lanelet2VectorMap/Namespaces/traffic_light_id` topic, check the checkbox, and then **reload** this topic by clicking the checkbox of `Map` twice to re-visualize the IDs (see the image below). You can have a closer look at the IDs by zooming in the region or by changing the View type.
+
+![see-traffic-light-ID](images/planning/lane-following/see-traffic-light-ID.png)
+
+And then you can select the ID of the desired traffic light from the `ID` _Combobox_. The image below shows a traffic light whose `ID` is 34836 and its color is set to `RED`.
+
+![set-traffic-light](images/planning/lane-following/set-traffic-light.png)
+
+Finally, click the `PUBLISH` button to send the traffic light status to the simulator. Any planned path that goes past the selected traffic light will then change accordingly.
+
+![send-traffic-light-color](images/planning/lane-following/send-traffic-light-color.png)
```

Comment:
- Reordered this section a lot, since I think it makes more sense to show the steps required to add the panel and set the ID and color before explaining how to find the IDs or troubleshooting the IDs not appearing.
- The `see-traffic-light-ID.png` seems to be used to show two different things (how to find the `traffic_light_id` topic for troubleshooting, and how to find the traffic light IDs on the map). I think it would be better to split the image so that we can clearly have one image to explain the troubleshooting and a different image to show how to find the IDs on the map. Any thoughts on this?
- Removed the description of the "set" screenshot since the screenshot is self-explanatory.
- For the Lane Driving tutorial, we have a screenshot showing how to add the AutowareStatePanel, so should we also add a similar screenshot showing how to add the TrafficLightPublishPanel?

```suggestion
a) Go to `Panels -> Add new panel`, select `TrafficLightPublishPanel`, and then press `OK`. 
b) In `TrafficLightPublishPanel`, set the `ID` and color of the traffic light.
c) Click the `SET` button.

![set-traffic-light](images/planning/lane-following/set-traffic-light.png)

d) Finally, click the `PUBLISH` button to send the traffic light status to the simulator. Any planned path that goes past the selected traffic light will then change accordingly.

![send-traffic-light-color](images/planning/lane-following/send-traffic-light-color.png)

By default, Rviz should display the ID of each traffic light on the map. You can have a closer look at the IDs by zooming in the region or by changing the View type.

In the event that the IDs are not displayed, try the following troubleshooting steps:
a) In the `Displays` panel, find the `traffic_light_id` topic by toggling the triangle icons next to `Map > Lanelet2VectorMap > Namespaces`. 
b) Check the `traffic_light_id` checkbox.
c) Reload the topic by clicking the `Map` checkbox twice.

![see-traffic-light-ID](images/planning/lane-following/see-traffic-light-ID.png)
```